scripts/merge-script/pull-request.ts

```diff
@@ -0,0 +1,128 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import * as Octokit from '@octokit/rest';
+import {GitClient} from './git';
+import {PullRequestMergeTask} from './index';
+import {matchesPattern} from './string-pattern';
+import {getBranchesFromTargetLabel, getTargetLabelFromPullRequest} from './target-label';
+
+/** Enum that describes different states of a pull request. */
+export const enum PullRequestStatus {
+  CLA_UNSIGNED,
+  NOT_MERGE_READY,
+  FAILING_CI_JOBS,
+  PENDING_CI_JOBS,
+  NO_TARGET_LABEL,
+  MERGE_CONFLICTS,
+  UNKNOWN_MERGE_ERROR,
+  UNSATISFIED_BASE_SHA,
+  NOT_FOUND,
+  MISMATCHING_TARGET_BRANCH,
+}
+
+/** Interface that describes a pull request. */
+export interface PullRequest {
+  /** Number of the pull request. */
+  prNumber: number;
+  /** Title of the pull request. */
+  title: string;
+  /** Labels applied to the pull request. */
+  labels: string[];
+  /** List of branches this PR should be merged into. */
+  targetBranches: string[];
+  /** Branch that the PR targets in the Github UI. */
+  githubTargetBranch: string;
+  /** Amount of commits this pull request contains. */
```

Comment:
nit, can we do this as `commitCount`?

Reply:
Haha 😂 You always get me with that one. Sure, will change it!